benchmarks/gabriel-benchmarks/takr.lisp

```diff
@@ -0,0 +1,1041 @@
+         (t (tak84 (tak8 (- x 1) y z)
+                   (tak24 (- y 1) z x)
+                   (tak28 (- z 1) x y)))))
+(defun tak84 (x y z)
+   (cond ((>= y x) z)
+         (t (tak85 (tak45 (- x 1) y z)
+                   (tak35 (- y 1) z x)
+                   (tak45 (- z 1) x y)))))
+(defun tak85 (x y z)
+   (cond ((>= y x) z)
+         (t (tak86 (tak82 (- x 1) y z)
+                   (tak46 (- y 1) z x)
+                   (tak62 (- z 1) x y)))))
+(defun tak86 (x y z)
+   (cond ((>= y x) z)
+         (t (tak87 (tak19 (- x 1) y z)
+                   (tak57 (- y 1) z x)
+                   (tak79 (- z 1) x y)))))
+(defun tak87 (x y z)
+   (cond ((>= y x) z)
+         (t (tak88 (tak56 (- x 1) y z)
+                   (tak68 (- y 1) z x)
+                   (tak96 (- z 1) x y)))))
+(defun tak88 (x y z)
+   (cond ((>= y x) z)
+         (t (tak89 (tak93 (- x 1) y z)
+                   (tak79 (- y 1) z x)
+                   (tak13 (- z 1) x y)))))
+(defun tak89 (x y z)
+   (cond ((>= y x) z)
+         (t (tak90 (tak30 (- x 1) y z)
+                   (tak90 (- y 1) z x)
+                   (tak30 (- z 1) x y)))))
+(defun tak90 (x y z)
+   (cond ((>= y x) z)
+         (t (tak91 (tak67 (- x 1) y z)
+                   (tak1 (- y 1) z x)
+                   (tak47 (- z 1) x y)))))
+(defun tak91 (x y z)
+   (cond ((>= y x) z)
+         (t (tak92 (tak4 (- x 1) y z)
+                   (tak12 (- y 1) z x)
+                   (tak64 (- z 1) x y)))))
+(defun tak92 (x y z)
+   (cond ((>= y x) z)
+         (t (tak93 (tak41 (- x 1) y z)
+                   (tak23 (- y 1) z x)
+                   (tak81 (- z 1) x y)))))
+(defun tak93 (x y z)
+   (cond ((>= y x) z)
+         (t (tak94 (tak78 (- x 1) y z)
+                   (tak34 (- y 1) z x)
+                   (tak98 (- z 1) x y)))))
+(defun tak94 (x y z)
+   (cond ((>= y x) z)
+         (t (tak95 (tak15 (- x 1) y z)
+                   (tak45 (- y 1) z x)
+                   (tak15 (- z 1) x y)))))
+(defun tak95 (x y z)
+   (cond ((>= y x) z)
+         (t (tak96 (tak52 (- x 1) y z)
+                   (tak56 (- y 1) z x)
+                   (tak32 (- z 1) x y)))))
+(defun tak96 (x y z)
+   (cond ((>= y x) z)
+         (t (tak97 (tak89 (- x 1) y z)
+                   (tak67 (- y 1) z x)
+                   (tak49 (- z 1) x y)))))
+(defun tak97 (x y z)
+   (cond ((>= y x) z)
+         (t (tak98 (tak26 (- x 1) y z)
+                   (tak78 (- y 1) z x)
+                   (tak66 (- z 1) x y)))))
+(defun tak98 (x y z)
+   (cond ((>= y x) z)
+        (t (tak99 (tak63 (- x 1) y z)
+                  (tak89 (- y 1) z x)
+                  (tak83 (- z 1) x y)))))
+(defun tak99 (x y z)
+  (cond ((>= y x) z)
+        (t (lisp-takr (lisp-takr (- x 1) y z)
+                      (lisp-takr (- y 1) z x)
+                      (lisp-takr (- z 1) x y)))))
+
+;;;
+;;;
+;;;
+
+
+(cl:in-package #:coalton-benchmarks/native)
+
+(cl:declaim (cl:optimize (cl:speed 3) (cl:safety 0)))
+
+(coalton-toplevel
```

Comment:
To be a reasonable benchmark every coalton function in this file should have type annotations.

Reply:
Sounds good, this afternoon's zen garden

Reply:
I added type annotations to every coalton and lisp function